var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -64,28 +68,30 @@ class Hdf5(AutotoolsPackage):
 
     variant('mpi', default=True, description='Enable MPI support')
     variant('szip', default=False, description='Enable szip support')
+    variant('zlib', default=True, description='Enable zlib support')
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
-    # Build HDF5 with API compaitibility.
-    variant('api', default='none', description='choose api compatibility', values=('v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
+    # Build HDF5 with API compatibility.
+    variant('api', default='none', description='Choose api compatibility', values=('v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
```

Comment:
I would ask someone who knows more about this type of variants. Shouldn't we put `none` to the list of `values` too? At least for maintainability reasons. For example, `spack info hdf5` does not list `none` among the allowed values but I can still run `spack install hdf5 api=none`.

Reply:
I don't think `none` even makes sense for this variant.  Shouldn't the default be `v112` or `v110`?

Reply:
"none" has been added to develop recently by others, along with I think a test to insure that all possible values are in the list of values.  "none" in this case means don't enable compatibility macros for versions earlier than the one built, and for 1.10.x "none" sets the api version to v110, for 1.12.x , "none" sets it to v112, and so on.  Perhaps "none" should be renamed, or the description revised?